homeassistant/components/sensor/nest.py

```diff
@@ -4,49 +4,44 @@
 For more details about this platform, please refer to the documentation at
 https://home-assistant.io/components/sensor.nest/
 """
-from itertools import chain
 import logging
 
-from homeassistant.components.nest import DATA_NEST, SIGNAL_NEST_UPDATE
-from homeassistant.helpers.dispatcher import async_dispatcher_connect
-from homeassistant.helpers.entity import Entity
+from homeassistant.components.nest import DATA_NEST, NestSensorDevice
 from homeassistant.const import (
     TEMP_CELSIUS, TEMP_FAHRENHEIT, CONF_MONITORED_CONDITIONS,
-    DEVICE_CLASS_TEMPERATURE)
+    DEVICE_CLASS_TEMPERATURE, DEVICE_CLASS_HUMIDITY)
 
 DEPENDENCIES = ['nest']
 
-SENSOR_TYPES = ['humidity',
-                'operation_mode',
-                'hvac_state']
+SENSOR_TYPES = ['humidity', 'operation_mode', 'hvac_state']
 
-SENSOR_TYPES_DEPRECATED = ['last_ip',
-                           'local_ip',
-                           'last_connection']
-
-DEPRECATED_WEATHER_VARS = {'weather_humidity': 'humidity',
-                           'weather_temperature': 'temperature',
-                           'weather_condition': 'condition',
-                           'wind_speed': 'kph',
-                           'wind_direction': 'direction'}
+TEMP_SENSOR_TYPES = ['temperature', 'target']
 
-SENSOR_UNITS = {'humidity': '%', 'temperature': '°C'}
+PROTECT_SENSOR_TYPES = ['co_status', 'smoke_status', 'battery_health']
 
-PROTECT_VARS = ['co_status', 'smoke_status', 'battery_health']
+STRUCTURE_SENSOR_TYPES = ['eta']
 
-PROTECT_VARS_DEPRECATED = ['battery_level']
+_VALID_SENSOR_TYPES = SENSOR_TYPES + TEMP_SENSOR_TYPES + PROTECT_SENSOR_TYPES \
+                      + STRUCTURE_SENSOR_TYPES
 
-SENSOR_TEMP_TYPES = ['temperature', 'target']
+SENSOR_UNITS = {'humidity': '%'}
 
-STRUCTURE_SENSOR_TYPES = ['eta']
+SENSOR_DEVICE_CLASSES = {'humidity': DEVICE_CLASS_HUMIDITY}
 
 VARIABLE_NAME_MAPPING = {'eta': 'eta_begin', 'operation_mode': 'mode'}
 
-_SENSOR_TYPES_DEPRECATED = SENSOR_TYPES_DEPRECATED \
-    + list(DEPRECATED_WEATHER_VARS.keys()) + PROTECT_VARS_DEPRECATED
+SENSOR_TYPES_DEPRECATED = ['last_ip',
+                           'local_ip',
+                           'last_connection',
+                           'battery_level']
```

Comment:
Is this a new deprecation? Does that make this a breaking change?

Reply:
No, it is old, I just rearrange the order of variables.